hawtio-util/src/test/java/io/hawt/util/FilesTest.java

```diff
@@ -0,0 +1,301 @@
+
+    when(file1.isDirectory()).thenReturn(false);
+    when(file1.delete()).thenReturn(true);
+    when(file21.isDirectory()).thenReturn(false);
+    when(file21.delete()).thenReturn(true);
+    when(file22.isDirectory()).thenReturn(false);
+    when(file22.delete()).thenReturn(true);
+    when(file2.listFiles()).thenReturn(new File[] {file21, file22});
+    when(file2.isDirectory()).thenReturn(true);
+    when(file2.delete()).thenReturn(true);
+    when(file3.isDirectory()).thenReturn(false);
+    when(file3.delete()).thenReturn(true);
+    when(file.isDirectory()).thenReturn(true);
+    when(file.listFiles()).thenReturn(new File[] {file1, file2, file3});
+    Assert.assertEquals(6, Files.recursiveDelete(file));
+  }
+
+  @Test
+  public void getRelativePath() throws Exception, IOException {
+    // Setup mocks
+    final File rootDir = mock(File.class);
+    final File file = mock(File.class);
+ 
+    // Arrange, Act & Assert result
+    when(rootDir.getCanonicalPath()).thenReturn("/root");
+    when(file.getCanonicalPath()).thenReturn("/tmp/file");
+    Assert.assertEquals("/tmp/file", Files.getRelativePath(rootDir, file));
+
+    when(file.getCanonicalPath()).thenReturn("/root/tmp/file");
+    Assert.assertEquals("/tmp/file", Files.getRelativePath(rootDir, file));   
+  }
+
+  @Test
+  public void readBytesException() throws FileNotFoundException, IOException {
+    // Act
+    thrown.expect(FileNotFoundException.class);
+    Files.readBytes(null);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @PrepareForTest({Files.class, java.nio.file.Files.class})
+  @Test
+  public void getMimeTypeException() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    mockStatic(java.nio.file.Files.class);
+
+    // Arrange
+    when(file.toPath()).thenReturn(Paths.get("/tmp/file"));
+    when(file.isDirectory()).thenReturn(false);
+    when(file.getName()).thenReturn("file");
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenThrow(new IOException());
+
+    // Act & Assert result
+    Assert.assertEquals("text/plain", Files.getMimeType(file));
+  }
+
+  @PrepareForTest({Files.class, java.nio.file.Files.class})
+  @Test
+  public void getMimeType() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    mockStatic(java.nio.file.Files.class);
+
+    // Arrange, Act & Assert result
+    when(file.toPath()).thenReturn(Paths.get("/tmp/file"));
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenReturn("application/octet-stream");
+    Assert.assertEquals("application/octet-stream", Files.getMimeType(file));
+
+    when(java.nio.file.Files.probeContentType(isA(Path.class))).thenReturn("   ");
+    when(file.isDirectory()).thenReturn(true);
+    Assert.assertEquals("application/zip", Files.getMimeType(file));
+
+    when(file.isDirectory()).thenReturn(false);
+    
+    when(file.getName()).thenReturn(".xml");
```

Comment:
I would give the returns of `.getName()` names (as well as extensions) like `"sample.xml"` to verify the usage of `endsWith` (and to make it clear that `getName` doesn't return the extension only)

Reply:
Done.